### PR TITLE
docs - some new and updated reference info for statistics

### PR DESCRIPTION
**Pull request by lisakowen**
bring some statistics-related doc reference content up to postgres v12.  in this PR:
- update descriptions for default_statistics_target and gp_statistics_pullup_from_child_partition gucs
- add sql reference pages for ALTER/CREATE/DROP STATISTICS
- update pg_statistic sys catalog ref page
- add system catalog ref pages for pg_statistic_ext, pg_statistic_ext_data, pg_stats, pg_stats_ext

question: on the pg_statistic ref page,  there is a greenplum-specific warning (search for XXX).  is this still valid?

this PR does NOT include "using" topic updates or additions for general or extended statistics.

doc review site links (behind vpn):
- ALTER STATISTICS:  https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Greenplum/sqlref-stats/greenplum-database/GUID-ref_guide-sql_commands-ALTER_STATISTICS.html
- pg_statistic_ext:  https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Greenplum/sqlref-stats/greenplum-database/GUID-ref_guide-system_catalogs-pg_statistic_ext.html
- default_statistics_target guc description (link location is off):  https://docs-staging.vmware.com/en/draft/VMware-Tanzu-Greenplum/sqlref-stats/greenplum-database/GUID-ref_guide-config_params-guc-list.html#debug_print_slice_table

### CODE REVIEW

File: gpdb-doc/markdown/ref_guide/system_catalogs/pg_statistic_ext.html.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+
+|column|type|references|description|
+|------|----|----------|-----------|
+|`stxrelid`|oid|[pg\_class](pg_class.html).oid | The table containing the columns described by this object |
+|`stxname`|name| | The name of the statistics object |
+|`stxnamespace`|[pg\_namespace](pg_namespace.html).oid| | The object identifier of the namespace that contains this statistics object |
+|`stxowner`|oid|[pg\_authid](pg_authid.html).oid | The owner of the statistics object |
+|`stxkeys`|int2vector|[pg\_attribute](pg_attribute.html).oid | An array of attribute numbers, indicating which table columns are covered by this statistics object; for example, a value of `1 3` would mean that the first and the third table columns are covered |
+|`stxkind`|char[]| | An array containing codes for the enabled statistics kinds; valid values are: `d` for n-distinct statistics, `f` for functional dependency statistics, and `m` for most common values \(MCV\) list statistics |
+
```
Comment (QingMa-gp) [inline gpdb-doc/markdown/ref_guide/system_catalogs/pg_statistic_ext.html.md:13]: We should add this paragraph. Refer to: https://www.postgresql.org/docs/12/catalog-pg-statistic-ext.html.
> The `pg_statistic_ext` entry is filled in completely during CREATE STATISTICS, but the actual statistical values are not computed then. Subsequent ANALYZE commands compute the desired values and populate an entry in the `pg_statistic_ext_data` catalog.

File: gpdb-doc/markdown/ref_guide/system_catalogs/pg_statistic_ext_data.html.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+Like `pg_statistic`, `pg_statistic_ext_data` should not be readable by the public, since the contents might be considered sensitive. \(Example: most common combinations of values in columns might be quite interesting.\) [pg\_stats\_ext](pg_stats_ext.html) is a publicly readable view on `pg_statistic_ext_data` \(after joining with `pg_statistic_ext`\) that only exposes information about those tables and columns that are readable by the current user.
+
+|column|type|references|description|
+|------|----|----------|-----------|
+|`stxoid`|oid|[pg\_statistic_ext](pg_statistic_ext.html).oid | The extended statistic object containing the definition for this data. |
+|`stxdndistinct`|pg\_ndistinct| | N-distinct counts, serialized as `pg_ndistinct` type. |
+|`stxddependencies`|pg\_dependencies| | Functional dependency statistics, serialized as `pg_dependencies` type. |
+|`stxdmcv`|pg\_mcv\_list|[pg\_authid](pg_authid.html).oid | MCV \(most-common values\) list statistics, serialized as `pg_mcv_list` type. |
```
Comment (QingMa-gp) [inline gpdb-doc/markdown/ref_guide/system_catalogs/pg_statistic_ext_data.html.md:12]: The reference to `stxdmcv` should be empty.

**Comment by lisakowen** [comment]
thank you for reviewing so quickly and catching these, @QingMa-gp .

**Comment by lisakowen** [comment]
@QingMa-gp - i've addressed your comments, can you take another look?

**Review by QingMa-gp — APPROVED** [review]
LGTM.

File: gpdb-doc/markdown/ref_guide/sql_commands/DROP_STATISTICS.html.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+DROP STATISTICS [ IF EXISTS ] <name> [, ...] [ CASCADE | RESTRICT ]
+```
+
+## <a id="section3"></a>Description
+
+`DROP STATISTICS` removes statistics object\(s\) from the database. Only the statistics object's owner, the schema owner, or a superuser can drop a statistics object.
+
+## <a id="section4"></a>Parameters 
+
+`IF EXISTS`
```
Comment (dyozie) [inline gpdb-doc/markdown/ref_guide/sql_commands/DROP_STATISTICS.html.md:17]: Just wanted to point out that formatting of literal parts of the Parameter listings isn't consistent. This seems to be the only entry that's formatted as a command.

File: gpdb-doc/markdown/ref_guide/system_catalogs/pg_statistic_ext.html.md [hunk 4]
```diff
@@ -0,0 +1,17 @@
+# pg_statistic_ext
+
+The `pg_statistic_ext` system catalog table holds definitions of extended planner statistics. Each row in this catalog corresponds to a *statistics object* created with [CREATE STATISTICS](../sql_commands/CREATE_STATISTICS.html).
+
+|column|type|references|description|
+|------|----|----------|-----------|
+|`stxrelid`|oid|[pg\_class](pg_class.html).oid | The table containing the columns described by this object |
```
Comment (dyozie) [inline gpdb-doc/markdown/ref_guide/system_catalogs/pg_statistic_ext.html.md:7]: The description entries here don't have punctuation, while it seems like punctuation is used for the other newly-added tables.

**Review by dyozie — APPROVED** [review]

File: gpdb-doc/markdown/ref_guide/system_catalogs/pg_statistic.html.md [hunk 5]
```diff
@@ -1,29 +1,30 @@
...
+`pg_statistic` should not be readable by the public, since even statistical information about a table's contents should be considered sensitive. \(Example: minimum and maximum values of a salary column\). [pg\_stats](pg_stats.html) is a publicly readable view on `pg_statistic` that only exposes information about those tables that are readable by the current user.
 
-**Warning:** Diagnostic tools such as `gpsd` and `minirepro` collect sensitive information from `pg_statistic`, such as histogram boundaries, in a clear, readable form. Always review the output files of these utilities to ensure that the contents are acceptable for transport outside of the database in your organization.
+XXX **Warning:** Diagnostic tools such as `gpsd` and `minirepro` collect sensitive information from `pg_statistic`, such as histogram boundaries, in a clear, readable form. Always review the output files of these utilities to ensure that the contents are acceptable for transport outside of the database in your organization.
```
Comment (dyozie) [inline gpdb-doc/markdown/ref_guide/system_catalogs/pg_statistic.html.md:13]: Does the XXX here mean it's not certain we need this warning?

File: gpdb-doc/markdown/ref_guide/sql_commands/DROP_STATISTICS.html.md (hunk 3 above)
Comment (lisakowen) [reply gpdb-doc/markdown/ref_guide/sql_commands/DROP_STATISTICS.html.md:17]: thanks for noticing the inconsistencies.  i will address them in the "using" docs PR.

we should probably do a formatting consistency check on the ref pages when they are all updated.

**Pull request merged**